pkg/grpc/api.go

```diff
@@ -661,7 +661,9 @@ func (a *api) SaveState(ctx context.Context, in *runtimev1pb.SaveStateRequest) (
 			Metadata: s.Metadata,
 		}
 
-		if req.Metadata[contribMetadata.ContentType] == contenttype.JSONContentType {
+		contentType, hasContentType := s.Metadata[contribMetadata.ContentType]
+
+		if hasContentType && contentType == contenttype.JSONContentType {
```

Comment:
Nit: `contentType` can't be the JSON one if there's no content type in the dictionary

```suggestion
		if contentType == contenttype.JSONContentType {
```